fs/localfs/shallow_fs.go

```diff
@@ -63,21 +63,21 @@ func WriteShallowPlaceholder(path string, de *snapshot.DirEntry) (string, error)
 	return mp, nil
 }
 
-func dirEntryFromPlaceholder(path string) (*snapshot.DirEntry, error) {
+func dirEntryFromPlaceholder(path string) (*snapshot.DirEntry, bool, error) {
```

Comment:
While the rationale seems reasonable, it also seems redundant. It seems that the `bool` return value would always be `false` when `error != nil`, and it could either be `true` or `false` depending on the first return value, most likely `false` when `v1 == nil`.

Does this pattern, and thus this linter add real value? what am I missing here?

Reply:
the pattern this enforces is:

value, err := someFunction()
if err != nil {
   // 
}

now value is assumed to be != nil.

By adding boolean we're making it explicit that there's more than one thing to look at (not just dirEntry, but also the boolean).